clang/include/clang/Basic/Builtins.td

```diff
@@ -3452,9 +3452,10 @@ def Fmod : FPMathTemplate, LibBuiltin<"math.h"> {
 
 def Frexp : FPMathTemplate, LibBuiltin<"math.h"> {
   let Spellings = ["frexp"];
-  let Attributes = [NoThrow];
+  let Attributes = [NoThrow, Constexpr];
```

Comment:
The `Constexpr` here makes the non-`__builtin_`-prefixed version of the function `constexpr` under all language levels.
I checked the code: The `OnlyBuiltinPrefixedAliasIsConstexpr` flag can _add_ `constexpr`-ness; it does not _restrict_ it.

For reference:
https://github.com/llvm/llvm-project/blob/702198fc9ac5dba392f9d9ba7c56467996343c0a/clang/utils/TableGen/ClangBuiltinsEmitter.cpp#L225-L226

We may need a new builtin attribute that expresses `constexpr` for the non-prefixed functions only for C++23 and up (but we might end up with C++26-and-up, etc. in the future). However, it seems making the non-prefixed functions actually non-`constexpr` (as opposed to simply failing to evaluate) is only a QoI concern (see https://github.com/llvm/llvm-project/pull/88978#issuecomment-2119080104 for more context).

@philnik777, thoughts?

Reply:
This also influences `__has_constexpr_builtin(fmin)`, which would return true but then fail to actually evaluate during a constant expression, which doesn't seem that great. IMO we'd probably want to introduce the concept of "constexpr since C++xy".

Reply:
> IMO we'd probably want to introduce the concept of "constexpr since C++xy".

@philnik777, are you envisioning something where `LangOptions` are passed into `Builtin::Context::isConstantEvaluated`?
https://github.com/llvm/llvm-project/blob/fb2c6597e39e9e1a775525ea0236b2f89e46acff/clang/include/clang/Basic/Builtins.h#L279

I think the existing `Constexpr`/"`E`" attribute can be left alone for use in "always `constexpr`" cases.

For the selectively-`constexpr` cases, `LibBuiltin` in `clang/include/clang/Basic/BuiltinsBase.td` should have `OnlyBuiltinPrefixedAliasIsConstexpr` renamed to `BuiltinPrefixedAliasIsConstexpr`. Then adding a new string property for `UnprefixedOnlyConstexprSince` makes sense to me. It can be encoded into `Builtin::Info` (and I think using the `LangFeatures` enumeration from `"clang/Basic/LangStandard.h"` is not terribly wrong).

Reply:
Done.